File: iso27001-springboot/src/main/java/com/iso27001/api/domain/users/events/UserCreatedEvent.java
package com.iso27001.api.domain.users.events;

/**
 * A.12 — Domain event emitted after user creation.
 * Contains only hashed email — raw PII is never included in events.
 */
public record UserCreatedEvent(
    String userId,
    String emailHash,   // SHA-256 of email — never raw email
    String role
) {}

File: iso27001-nestjs/src/domain/users/events/user-created.event.ts
/** A.12: Domain event emitted after a user is successfully created. */
export class UserCreatedEvent {
  constructor(
    public readonly userId: string,
    /** A.12: Never log raw email — use its SHA-256 hash */
    public readonly emailHash: string,
    public readonly role: string,
  ) {}
}
